frontend/app_flowy/lib/startup/tasks/application_widget.dart

```diff
@@ -39,31 +39,48 @@ class ApplicationWidget extends StatelessWidget {
   }) : super(key: key);
 
   @override
-  Widget build(BuildContext context) {
-    const ratio = 1.73;
-    const minWidth = 800.0;
-    setWindowMinSize(const Size(minWidth, minWidth / ratio));
-    // const launchWidth = 1310.0;
-    // setWindowFrame(const Rect.fromLTWH(0, 0, launchWidth, launchWidth / ratio));
+  Widget build(BuildContext context) => ChangeNotifierProvider(
+      create: (context) => ThemeProvider(),
+      builder: (context, _) {
+        const ratio = 1.73;
+        const minWidth = 800.0;
+        setWindowMinSize(const Size(minWidth, minWidth / ratio));
+        // const launchWidth = 1310.0;
+        // setWindowFrame(const Rect.fromLTWH(0, 0, launchWidth, launchWidth / ratio));
 
-    var _dark = ThemeType.dark;
-    var _light = ThemeType.light;
-    final theme = AppTheme.fromType(ThemeType.light);
-    theme.isDark = true;
-    return Provider.value(
-      value: theme,
-      child: MaterialApp(
-        builder: overlayManagerBuilder(),
-        debugShowCheckedModeBanner: false,
-        theme: theme.themeData,
-        localizationsDelegates: context.localizationDelegates,
-        supportedLocales: context.supportedLocales,
-        locale: context.locale,
-        navigatorKey: AppGlobals.rootNavKey,
-        home: child,
-      ),
-    );
+        AppTheme theme = AppTheme.fromType(ThemeType.light);
+
+        final themeProvider = Provider.of<ThemeProvider>(context);
+
+        return Provider.value(
+          value: theme,
+          child: MaterialApp(
+            builder: overlayManagerBuilder(),
+            debugShowCheckedModeBanner: false,
+            themeMode: themeProvider.themeMode,
+            theme: ThemeSwitch.light,
+            // theme: themeProvider.isDarkMode ? ThemeSwitch.dark : ThemeSwitch.light,
+            darkTheme: ThemeSwitch.dark,
+            localizationsDelegates: context.localizationDelegates,
+            supportedLocales: context.supportedLocales,
+            locale: context.locale,
+            navigatorKey: AppGlobals.rootNavKey,
+            home: child,
+          ),
+        );
+      });
+}
+
+class App_Theme extends ChangeNotifier {
+  ThemeType get theme => _theme;
+  ThemeType _theme = ThemeType.light;
+
+  set theme(ThemeType value) {
+    _theme = value;
```

Comment:
Compare the value is the same as _theme would be better

Reply:
Forgot to delete that, will review it maybe could use it somewhere.